packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,225 @@
+
+| Prop      | Type                                     | Default     | Description                                | Required |
+| --------- | ---------------------------------------- | ----------- | ------------------------------------------ | -------- |
+| `actions` | `{ primary: Action, secondary: Action }` | `undefined` | Renders a primary/secondary action buttons |          |
+
+```ts
+type Action = {
+  onClick: () => void;
+  text: string;
+  type?: 'button' | 'reset' | 'submit';
+  accessibilityLabel: string;
+  isLoading: boolean;
+  isDisabled: boolean;
+  icon: React.ReactNode;
+  iconPosition: 'left' | 'right';
+};
+```
+
+## Composition with DropDown
+
+We will export `BottomSheet` component separately as an independant component but generally the pattern will be to use it with SelectInput, where in mobile devices the Select's dropdown will be replaced by the BottomSheet.
+
+There are two approaches to doing it:
+
+1. We coupled the BottomSheet & SelectInput tightly and internally conditionally switch the components
+
+Pros:
+
+- Easy implementation for user's end
+- User's don't have to think about breakpoints, conditional rendering etc
+- From blade side, we will have greater control over the pattern
+
+Cons:
+
+- Bundle size will be impacted, even if users are desktop they will get the bundle of BottomSheet (vice versa)
+
+2. We expose BottomSheet indepandantly and let users lazy load the component as needed
+
+Pros:
+
+- No uneccesary bundle size impact for any of the platforms
+
+Cons:
+
+- Not trivial to implement from user's end, they will have to compose the BottomSheet & Select as per their needs.
+
+
+Considering the bundle size downside to approach 1, we decided to go ahead with approach 2.
+
+### Composition Example
+
+```jsx
+const BottomSheet = React.lazy();
+const DropdownOverlay = React.lazy();
+
+const SelectContent = () => {
+  return (
+    <>
+      <ActionListItem
+        leading={<ActionListItemIcon icon={SettingsIcon} />}
+        title="Settings"
+        value="settings"
+      />
+      <ActionListItem leading={<ActionListItemIcon icon={InfoIcon} />} title="Info" value="info" />
+    </>
+  );
+};
+
+const App = () => {
+  const [open, setOpen] = useState(false);
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" onClick={() => setOpen(true)} />
+      {isMobile ? (
+        <BottomSheet open={open} onDismiss={() => setOpen(false)}>
+          <BottomSheetHeader>
+            <BottomSheetHeaderLeading title="Payments Links" prefix={PayIcon} />
+          </BottomSheetHeader>
+          <BottomSheetBody>
+            <SelectContent />
+          </BottomSheetBody>
+          <BottomSheet>
+            <BottomSheetLeading title="Footer Title" prefix={icon} />
+            <BottomSheetTrailing
+              actions={{
+                primaryAction: { text: 'Confirm' },
+                secondaryAction: { text: 'Close' },
+              }}
```

Comment:
Can we do something like this?

```jsx
<ResponsiveMenuOverlay>
  <BottomSheetHeader /> // we hide this desktop
  <ActionList>
    <ActionListItem />
  </ActionList>
  <BottomSheetFooter /> // hide this in desktop
</ResponsiveMenuOverlay>
```

Reply:
And basically somehow do lazy-loading on our end 🤔 ?

Reply:
> And basically somehow do lazy-loading on our end 🤔 ?

Not feasible to do lazy loading our end as mentioned in the composition section.

Reply:
oh yup saw your comments on slack. Makes sense.